packages/web3-wrapper/src/web3_wrapper.ts

```diff
@@ -489,6 +495,21 @@ export class Web3Wrapper {
     public async setHeadAsync(blockNumber: number): Promise<void> {
         await this._sendRawPayloadAsync<void>({ method: 'debug_setHead', params: [this._web3.toHex(blockNumber)] });
     }
+    /**
+     * Returns either NodeType.Geth or NodeType.Ganache depending on the type of
+     * the backing Ethereum node. Throws for any other type of node. This
+     * function caches the result and so subsequent calls are fast.
```

Comment:
I don't see any caching going on here, and there shouldn't be any. Please remove this sentence.

Reply:
I meant to add caching here (just did so in the most recent commit). I think we should cache this. This is a function that gets called more than a thousand times and the `NodeType` never changes for a given instance of `Web3Wrapper`. I just checked and it does improve performance by a few seconds (not a huge difference but it's 5 lines of code so probably worth it).

Reply:
For context: Running tests against Geth went from ~60s to ~53s and Ganache went from ~81s to ~78s.